yoyodyne/schedulers.py

```diff
@@ -108,6 +115,55 @@ def __repr__(self) -> str:
         )
 
 
+class ReduceOnPlateau(optim.lr_scheduler.ReduceLROnPlateau):
+    """Reduce on plateau scheduler."""
+
+    def __init__(
+        self,
+        optimizer,
+        reduceonplateau_mode,
+        reduceonplateau_factor,
+        reduceonplateau_patience,
+        min_lr,
+        **kwargs,
+    ):
+        """Initializes the LR scheduler.
+
+        Args:
+            optimizer (optim.Optimizer): optimizer.
+            reduceonplateau_mode (str): Determines whether to reduce the LR
+                when the metric stops decreasing (`loss`) or increasing
+                (`accuracy`).
+            reduceonplateau_factor (float): Factor by which the
+                learning rate will be reduced. new_lr = lr * factor.
+            reduceonplateau_patience (int): Number of epochs with no
+                improvement before reducing LR.
+            min_lr (float): Lower bound on the learning rate.
+            **kwargs: ignored.
+
+        Currently lets the following hyperparameters use the pytorch defaults:
+            threshold,
+            threshold_mode,
+            cooldown,
+            eps,
+        """
+        super().__init__(
+            optimizer,
+            mode="min" if reduceonplateau_mode == "loss" else "max",
+            factor=reduceonplateau_factor,
+            patience=reduceonplateau_patience,
+            min_lr=min_lr,
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"{self.optimizer}, {self.mode}, "
+            f"{self.factor}, {self.min_lrs}), "
+            f"{self.patience}"
```

Comment:
Possibly dumb question but do you mean to have that closed parentheses before the end? This will look like `ReduceOnPlateau(Adam, "loss", .1, 0), 5`.

Reply:
Oh no, that was a mistake good catch.